fix(android)(9_0_X): app/module builds fail with JDK 13

jquick-axway opened the pull request:
**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-27778
- https://jira.appcelerator.org/browse/TIMOB-27780
- https://jira.appcelerator.org/browse/TIMOB-27781

**Cherry-pick of PR:** #11507

jquick-axway committed:
chore(android): update gradle build tools to 3.6.0

- Added "android.useAndroidX" gradle property which is now requried or else a build failure will occur.
- Resolved Java and C/C++ build order issue with newest tool.

jquick-axway committed:
fix(android): updated gradle wrapper so it can build with JDK 13

- Updated gradle wrapper/library to version 6.2.1. (Gradle added JDK 13 support in 6.0.0.)

build commented:
<!--
  0 failure: 
  1 warning:  This PR has miles...
  3 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR has milestone set to 9.0.1, but the version defined in package.json is 9.0.0
Please either:
- Update the milestone on the PR
- Update the version in package.json
- Hold the PR to be merged later after a release and version bump on this branch</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-11510/1/artifact/dist/mobilesdk-9.0.0.v20200228211314-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 6555 tests are passing.
(There are 703 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cb4ad655cfb39b4c26fd73cb755cd2c227949c01
</p>

jquick-axway committed:
fix(android): hyperloop builds fail with JDK 12 or higher

- JDK 12 dropped support for Java 6 language support. Hyperloop now compiles with Java 8 language.
- Updated hyperloop module version from 5.0.1 to 5.0.2.

garymathews committed:
Merge branch '9_0_X' into TIMOB-27778-9_0_X

ssaddique committed:
Merge branch '9_0_X' into TIMOB-27778-9_0_X

ssaddique committed:
Merge branch '9_0_X' into TIMOB-27778-9_0_X

ssaddique committed:
Merge branch '9_0_X' into TIMOB-27778-9_0_X

ssaddique commented:
FR: Pass.
Please see [this JIRA ticket](https://jira.appcelerator.org/browse/TIMOB-27781) for details.
